includes/Core/Permissions/Permissions.php

```diff
@@ -360,6 +360,21 @@ private function map_meta_capabilities( array $caps, $cap, $user_id, $args ) {
 			$caps = array_merge( $caps, $this->check_dashboard_sharing_capability( $cap, $user_id, $args ) );
 		}
 
+		// Handle VIEW_SPLASH capability.
+		if ( self::VIEW_SPLASH === $cap ) {
+			$caps = array_merge( $caps, $this->check_view_splash_capability( $user_id ) );
+		}
+
+		// Handle VIEW_AUTHENTICATED_DASHBOARD capability.
+		if ( self::VIEW_AUTHENTICATED_DASHBOARD === $cap ) {
+			$caps = array_merge( $caps, $this->check_view_authenticated_dashboard_capability( $user_id ) );
+		}
+
+		// Special Handling of VIEW_DASHBOARD and VIEW_POSTS_INSIGHTS capabilities when the dashboardSharing feature flag is enabled.
+		if ( Feature_Flags::enabled( 'dashboardSharing' ) && in_array( $cap, array( self::VIEW_DASHBOARD, self::VIEW_POSTS_INSIGHTS ), true ) ) {
+			$caps = array_merge( $caps, $this->check_view_dashboard_capability( $user_id ) );
+		}
```

Comment:
We can remove the feature flag check here and instead have it just around the `check_view_shared_dashboard_capability` call within this method, since only that is the part that should be conditional to the `dashboardSharing` feature flag.

However, your current approach works too, since both `VIEW_DASHBOARD` and `VIEW_POSTS_INSIGHTS` are already covered by the above clause starting in line 342. So either works for me.